core/src/java/show/ezkz/casprzak/core/tools/functiongenerators/TaylorSeries.java

```diff
@@ -0,0 +1,120 @@
+import show.ezkz.casprzak.core.functions.commutative.Sum;
+import show.ezkz.casprzak.core.functions.endpoint.Constant;
+import show.ezkz.casprzak.core.functions.endpoint.Variable;
+import show.ezkz.casprzak.core.tools.MiscTools;
+import show.ezkz.casprzak.core.tools.VariableTools;
+
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Set;
+
+/**
+ * The methods in {@link TaylorSeries} produce Taylor Series for functions.
+ */
+public class TaylorSeries {
+
+    private TaylorSeries(){}
+
+    /**
+     * Returns a Maclaurin series of the specified function of a given degree
+     * @param function the function whose Maclaurin series is being found
+     * @param degree the degree of the Maclaurin polynomial
+     * @return a Maclaurin series of the specified function
+     */
+    public static GeneralFunction makeTaylorSeries(GeneralFunction function, int degree) {
+        return makeTaylorSeries(function, degree, 0);
+    }
+
+	/**
+	 * Returns a Maclaurin series of the specified function of a given degree
+	 * @param function the function whose Maclaurin series is being found
+	 * @param degree the degree of the Maclaurin polynomial
+	 * @param variable the String of the {@link Variable} of the polynomial
+	 * @return a Maclaurin series of the specified function
+	 */
+	public static GeneralFunction makeTaylorSeries(GeneralFunction function, int degree, String variable) {
+		Set<String> variables = VariableTools.getAllVariables(function);
+		if (variables.size() > 1)
+			throw new UnsupportedOperationException("Cannot perform taylor series with a function of more than 1 variable.");
+
+		if (Settings.singleVariableDefault.equals(variable))
+			return makeTaylorSeries(function, degree);
+
+		Map<String, Variable> reverseSubstitution = new HashMap<>() {
+			{
+				put(variable, new Variable(Settings.singleVariableDefault));
+			}
+		};
+		GeneralFunction taylorSeries = makeTaylorSeries(function.substituteVariables(reverseSubstitution), degree);
+		Map<String, Variable> substitution = new HashMap<>() {
+			{
+				put(Settings.singleVariableDefault, new Variable(variable));
+			}
+		};
+		return taylorSeries.substituteVariables(substitution);
+	}
+
+	/**
+	 * Returns a Taylor series of the specified function at the specified center of a given degree
+	 * @param function the function whose Taylor series is being found
+	 * @param degree the degree of the Taylor polynomial
+	 * @param center where the Taylor series is centered
+	 * @param variable the String of the {@link Variable} of the polynomial
+	 * @return a Taylor series of the specified function
+	 */
+	public static GeneralFunction makeTaylorSeries(GeneralFunction function, int degree, double center, String variable) {
+		Set<String> variables = VariableTools.getAllVariables(function);
+		if (variables.size() > 1)
+			throw new UnsupportedOperationException("Cannot perform taylor series with a function of more than 1 variable.");
```

Comment:
I also changed a ton of uncaptilazied laguerre / whatever names, please capitalize your names